### PR TITLE
Update HoughCircle_Demo.cpp

### PR DESCRIPTION
**Pull request by yyyhssss**
### This pullrequest changes

in line 94

int key is cannot used to compare with char

### CODE REVIEW

File: samples/cpp/tutorial_code/ImgTrans/HoughCircle_Demo.cpp [hunk 1]
```diff
@@ -91,7 +91,7 @@ int main(int argc, char** argv)
     // and refresh the content of the output image
     // until the user presses q or Q
     int key = 0;
-    while(key != 'q' && key != 'Q')
+    while((char) key != 'q' && (char) key != 'Q')
```
Comment (alalek) [inline samples/cpp/tutorial_code/ImgTrans/HoughCircle_Demo.cpp]: You should not truncate key value. This range exceed values of standard char "[-128;127]".
```
(char)0x131 == '1' /*0x31*/
```
is still "true", but it is wrong.
You should cast both to bigger value ("int" in this case) or try to change compiler options and disable noise warnings.

**Comment by StevenPuttemans** [comment]
Why would this code not run? If you really not okay with char ASCII value comparison to the int returned by `waitKey()` then you can simply fix this by changing the line ` while(key != 'q' && key != 'Q')` to ` while(key != ord('q') && key != ord('Q')` and then your warnings are all gone.

**Comment by StevenPuttemans** [comment]
Actually this sample has another issue, which is worse then the one you pointed out. [Line 105](https://github.com/yyyhssss/opencv/blob/22ee03eb49c48b57343db3036b7a0c512c31af28/samples/cpp/tutorial_code/ImgTrans/HoughCircle_Demo.cpp#L105) uses the function `waitKey(10)`. However in order to ensure that in every system and in every case the exact same value is returned, this should be changed to `waitKey(10) & 0xFF`. 

Or is this not needed anymore @alalek? I know it was in the past.

**Comment by alalek** [comment]
@StevenPuttemans You are right! `waitKey(10) & 0xFF` is still required.
In this case my note is almost wrong, except that "char" range is [-128, 127] and "key" range is [0; 255]. 

`cv::waitKey()` returns some key codes which are depends on UI backend. And their implementations are not equivalent unfortunately.

**Comment by sturkmen72** [comment]
https://github.com/opencv/opencv/pull/7866 already fixed this

**Pull request closed without merging**